services/metadata/src/metadata/persistence/permanent_id_requests.clj

```diff
@@ -0,0 +1,200 @@
+            [:statuses.date_assigned :status_date]
+            [:statuses.updated_by])
+    (join [:permanent_id_request_types :types] {:types.id :r.type})
+    (join [:permanent_id_request_statuses :statuses] {:statuses.permanent_id_request :r.id})
+    (join [:permanent_id_request_status_codes :status_codes]
+          {:status_codes.id :statuses.permanent_id_request_status_code})
+    (where-if-defined {:requested_by user})
+    (order :statuses.date_assigned :ASC)))
+
+(defn- list-permanent-id-requests-query
+  "Lists the Permanent ID Requests that have been submitted by the user."
+  [{user       :user
+    row-offset :offset
+    row-limit  :limit
+    sort-field :sort-field
+    sort-dir   :sort-dir}]
+  (subselect [(list-permanent-id-requests-subselect user) :reqs]
+      (fields :id :type :target_id :target_type :requested_by
+              [(sqlfn :first :status_date) :date_submitted]
+              [(sqlfn :last :status) :status]
+              [(sqlfn :last :status_date) :date_updated]
+              [(sqlfn :last :updated_by) :updated_by])
+      (group :id :type :target_id :target_type :requested_by)
+      (order (or sort-field :date_submitted) (or sort-dir :ASC))
+      (limit row-limit)
+      (offset row-offset)))
+
+(defn- format-request-listing
+  [request]
+  (-> request
+      (update-in [:date_submitted] millis-from-timestamp)
+      (update-in [:date_updated] millis-from-timestamp)
+      remove-nil-values))
+
+(defn list-permanent-id-requests
+  "Lists Permanent ID Requests, filtered and sorted by the given params."
+  [{:keys [statuses] :as params}]
+  (let [status-clause (when statuses ['in statuses])]
+    (map format-request-listing
+      (select [(list-permanent-id-requests-query params) :request_list]
+        (where-if-defined {:status status-clause})))))
+
+(defn get-permanent-id-request
+  "Gets Permanent ID Request details. If the given user is not nil, only fetches details if the request
+  was submitted by the given user."
+  [user request-id]
+  (first
+    (select [:permanent_id_requests :r]
+      (fields :r.id
+              :types.type
+              :target_id
+              :target_type
+              :requested_by)
+      (join [:permanent_id_request_types :types] {:types.id :r.type})
+      (where-if-defined {:r.id request-id
+                         :requested_by user}))))
+
+(defn get-most-recent-status
+  "Gets the most recent status for a Permanent ID Request."
+  [request-id]
+  ((comp :name first)
+    (select [:permanent_id_requests :r]
+            (fields :status_codes.name)
+            (join [:permanent_id_request_statuses :statuses]
+                  {:statuses.permanent_id_request :r.id})
+            (join [:permanent_id_request_status_codes :status_codes]
+                  {:status_codes.id :statuses.permanent_id_request_status_code})
+            (where {:r.id request-id})
+            (order :statuses.date_assigned :DESC)
+            (limit 1))))
```

Comment:
I usually leave out the top-level table (`permanent_id_requests` in this case) in these types of queries, but the effect is equivalent. I do it just to remove one of the join levels, which is probably done by the query optimizer anyway. The upshot is that no change is required, but I normally do something slightly different. :grinning: